internal/services/providers/kops/kops_test.go

```diff
@@ -0,0 +1,260 @@
+package kops
+
+import (
+	"context"
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/golang/mock/gomock"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/utils/pointer"
+
+	"castai-agent/internal/castai"
+	mock_castai "castai-agent/internal/castai/mock"
+	"castai-agent/internal/config"
+	mock_awsclient "castai-agent/internal/services/providers/eks/client/mock"
+	"castai-agent/internal/services/providers/gke"
+	"castai-agent/internal/services/providers/types"
+	"castai-agent/pkg/labels"
+)
+
+func TestProvider_RegisterCluster(t *testing.T) {
+	t.Run("autodiscover cluster properties", func(t *testing.T) {
+		require.NoError(t, os.Setenv("API_KEY", "123"))
+		require.NoError(t, os.Setenv("API_URL", "test"))
+
+		t.Cleanup(config.Reset)
+		t.Cleanup(os.Clearenv)
+
+		var objects []runtime.Object
+
+		namespaceID := uuid.New()
+
+		objects = append(objects, &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:  k8stypes.UID(namespaceID.String()),
+				Name: metav1.NamespaceSystem,
+				Annotations: map[string]string{
+					"addons.k8s.io/core.addons.k8s.io": `{"version":"1.4.0","channel":"s3://test-kops/test.k8s.local/addons/bootstrap-channel.yaml","manifestHash":"3ffe9ac576f9eec72e2bdfbd2ea17d56d9b17b90"}`,
```

Comment:
does this work the same for older and newer kops versions?

Reply:
Yes! Tested from 1.11 to 1.20.